lmdeploy/pytorch/nn/moe.py

```diff
@@ -89,23 +105,33 @@ def create_weights(self, hidden_dim: int, ffn_dim: int, num_experts: int,
                                    device=device)
         return gate_up_weights, down_weights
 
+    def create_weights_ep(self, hidden_dim: int, ffn_dim: int,
```

Comment:
`create_weights_tp` and `create_weights_ep` are actually the same

Reply:
https://github.com/InternLM/lmdeploy/blob/1f3038046778c15a3262fee1294f394ef0de6fe3/lmdeploy/pytorch/backends/cuda/moe.py#L36 expert list is not the origin experts

Reply:
I know. I mean
```
 gate_up_weights, down_weights = self.create_weights_ep(
                hidden_dim, ffn_dim, expert_list, dtype=dtype, device=device)
```
equals to 
```
gate_up_weights, down_weights = self.create_weights_tp(
                hidden_dim, ffn_dim, len(expert_list), dtype=dtype, device=device)
```
So only keeping one function is fine.

Reply:
expert_list could be None if ep is disabled.